manager/scheduler/scheduler.go

```diff
@@ -143,17 +143,21 @@ func (s *Scheduler) Run(ctx context.Context) error {
 		case event := <-updates:
 			switch v := event.(type) {
 			case api.EventCreateTask:
-				pendingChanges += s.createTask(ctx, v.Task)
+				s.createTask(ctx, v.Task)
+				tickRequired = true
 			case api.EventUpdateTask:
-				pendingChanges += s.updateTask(ctx, v.Task)
+				s.updateTask(ctx, v.Task)
+				tickRequired = true
 			case api.EventDeleteTask:
 				s.deleteTask(ctx, v.Task)
+				// deleting tasks may free up node resource, pending tasks should be re-evaluated.
+				tickRequired = true
```

Comment:
I'd change `deleteTask` to return true if `removeTask` returned true. Then we would not trigger a tick if the task that's being deleted was not scheduled.